lib/logger.js

```diff
@@ -636,11 +651,27 @@ class Logger extends EventEmitter {
       axios(config)
         .then((response) => {
           // We have a 200-level success code.
-          const totalLinesSent = buffer.length
+          let totalLinesSent = buffer.length
+
+          if (response.status === PARTIAL_SUCCESS_CODE) {
+            const statusCodes = response.data.status
```

Comment:
@darinspivey, can you check the existence of `response.data` just to increase the safety?

Reply:
If we did that, we'd be coding for a developer error.  It's not possible for `data` to be missing (unless you mock a test, and we don't do that).  At worst, it would be an empty string, which would make accessing `status` throw.  However, if that was the case, it means we've done something wrong at the ingester and are not returning the right body.

I'll meet you in the middle and do `const statusCodes = response.data.status || []`, but I try to avoid that.  It can misleading to someone in that it looks like a possible scenario when really it shouldn't be.

Reply:
Added fallback to `[]` for the codes. This approach has code smell because it means the `totalLinesSent` will not be able to remove the lines that failed from the count.  Typically, we should not code such situations, but being that this has direct customer impact that isn't quickly fixable, let's do the graceful approach.

Reply:
@darinspivey, I see that `response.data` is something `axios` is providing but in the case of not having `response.data` or having `response.data` as `undefined`, trying to get `response.data.status` is gonna throw an `Error` - I know we rely on `axios` and our ingester but this is for avoiding potential exceptions that might happen in the case of some changes in `axios` and/or our ingester.

Reply:
I agree with the graceful middle-ground here for the existence of `status`. We shouldn't write code to preemptively handle major changes to a library like `axios` though. If `response.data` is guaranteed to be there from current-version-of-axios it doesn't need to be checked.

Reply:
Axios will populate `data` with an empty string if *only* a 207 error code is returned (which is the bare minimum to hit this code path), so it won't throw, it'll just be `undefined`.

If we change away from axios, the name `data` will change and this code will be updated anyway.  We have all the proper tests to keep us from making those mistakes.  Don't worry :]